src/ready.rs

```diff
@@ -71,6 +75,76 @@ where
         Ok(fifo_count as usize / 20)
     }
 
+    #[cfg(feature = "async")]
+    pub async fn read_sample(&mut self) -> Result<Sample, Error<SPI::Error>> {
+        // Read one extra packet (discarded) in case the user has been slow in
+        // reading out samples.
+        // We read INT_STATUS, FIFO_COUNT_H, FIFO_COUNT_L, and then the data in
+        // one go. The leading byte in the buffer is used to signal register
+        // address, it's output doesn't contain data on return.
+        let mut buffer = [0u8; 44];
+        buffer[0] = crate::register_bank::bank0::INT_STATUS::ID | 0x80;
+        self.ll.bus.transfer_in_place(&mut buffer).await?;
+        let p = bytemuck::from_bytes::<FifoPacket4>(&buffer[4..24]);
+        Self::sample_from_packet4(p).ok_or(Error::NoSampleRead)
+    }
+
+    #[cfg(not(feature = "async"))]
+    pub fn read_sample(&mut self) -> Result<Sample, Error<SPI::Error>> {
+        // Read one extra packet (discarded) in case the user has been slow in
```

Comment:
This should not be a supported use case. If the user is reading too slow, the FIFO will overflow and we should (properly) fail, i.e. return an Error signaling this condition, or even panic.

Reply:
Yeah but in a real time system we never want FIFO overflow. Allowing this will make debugging significantly harder due to data inconsistencies. I suggested the change below :)